### PR TITLE
Detailed Build Guide

### PR DESCRIPTION
**Pull request by t0mmysm1th**
This guide intentionally avoids the use of Docker (see `Makefile`) due to errors when building directly on the raspi.

### CODE REVIEW

**Comment by dr-bonez** [comment]
I just want to note, this hasn't been forgotten, I've looked over it and it looks good. But I haven' had the time to test it yet.

I'll give this some proper attention when we gear up for 0.3.

**Comment by LochRED** [comment]
Hi,

I am getting an error when I run make. Not sure whats going wrong? Using a Pi 4 with BCM2711

pi@raspberrypi:~/embassy-os $ sudo make
docker run --rm -it -v ~/.cargo/registry:/root/.cargo/registry -v "/home/pi/embassy-os":/home/rust/src start9/rust-arm-cross:latest sh -c "(cd appmgr && cargo build --release --features=production)"
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm/v7) and no specific platform was requested
standard_init_linux.go:219: exec user process caused: exec format error
make: *** [Makefile:30: appmgr/target/armv7-unknown-linux-gnueabihf/release/appmgr] Error 1

**Comment by t0mmysm1th** [comment]
Please note that you would need to replace the `Makefile` and `make_image.sh` with the ones in my repo.

**Review by k0gen — APPROVED** [review]

File: BuildGuide.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      stack install --stack-yaml=stack-ghc-84.yaml --system-ghc
+      export PATH=~/.local/bin:$PATH
+      ```
+
+5. Clone EmbassyOS & try to *make* the `agent`:
+   1. First attempt
+      > :information_source: The first time you run **make** you'll get an error
+      
+      ```
+      sudo apt install llvm-9 libgmp-dev
```
Comment (elvece) [inline BuildGuide.md:76]: fwiw I needed to run `export PATH=/usr/lib/llvm-9/bin:$PATH` to resolve errors here.

**Comment by dr-bonez** [comment]
Closing in favor of #252

**Pull request closed without merging**